models/m_payment_mollie_api.erl

```diff
@@ -137,65 +147,240 @@ payment_url(MollieId) ->
+                    case api_call(get, "payments/" ++ z_convert:to_list(ExtId), [], Context) of
+                        {ok, #{ <<"resource">> := <<"payment">> } = JSON} ->
+                            m_payment_log:log(
+                                PaymentId,
+                                <<"WEBHOOK">>,
+                                [
+                                   {psp_module, mod_payment_mollie},
+                                   {description, "New webhook payment info"},
+                                   {payment, JSON}
+                                ],
+                                Context),
+                            handle_new_payment(PaymentId, Payment, JSON, Context);
+                        {error, Error} ->
+                            %% Log an error with the payment
+                            m_payment_log:log(
+                                PaymentId,
+                                <<"ERROR">>,
+                                [
+                                    {psp_module, mod_payment_mollie},
+                                    {description, "API Error fetching status from Mollie"},
+                                    {request_result, Error}
+                                ],
+                                Context),
+                            lager:error("API error creating mollie payment for #~p: ~p", [PaymentId, Error]),
+                            Error
+                    end;
+                PSP ->
+                    lager:error("Payment PSP Mollie webhook call for unknown PSP ~p / ~p: ~p", [PaymentNrOrId, ExtId, PSP]),
+                    {error, notfound}
+            end;
+        {error, notfound} ->
+            lager:error("Payment PSP Mollie webhook call with unknown id ~p / ~p", [PaymentNrOrId, ExtId]),
+            {error, notfound};
+        {error, _} = Error ->
+            lager:error("Payment PSP Mollie webhook call with id ~p / ~p, fetching payment error: ~p", [PaymentNrOrId, ExtId, Error]),
             Error
     end.
 
+handle_new_payment(PaymentId, Payment, #{ <<"recurringType">> := <<"recurring">> } = JSON, Context) ->
+    % A recurring payment for an existing payment.
+    % The given Payment MUST have 'recurring' set.
+    #{
+        <<"id">> := ExtId,
+        <<"status">> := Status,
+        <<"subscriptionId">> := _SubscriptionId,
+        <<"amount">> := Amount,
+        <<"description">> := Description
+    } = JSON,
+    % Create a new payment, referring to the PaymentId
+    Currency = proplists:get_value(currency, Payment),
+    AmountNr = z_convert:to_float(Amount),
+    DateTime = z_convert:to_datetime( status_date(JSON) ),
+    case m_payment:get_by_psp(mod_payment_mollie, ExtId, Context) of
+        {ok, Payment} ->
+            PaymentId = proplists:get_value(id, Payment),
+            update_payment_status(PaymentId, Status, DateTime, Context);
```

Comment:
Should `m_payment:update_pdp_handler` also be called here, as on line 313?

Reply:
Here the PSP is already correctly registered, so we don't need to update those fields.
On line 313 we add a new payment, and need to update that with the correct PSP.